include/eve/detail/function/simd/common/load.hpp

```diff
@@ -164,105 +68,189 @@ namespace eve::detail
   }
 
   //================================================================================================
-  // Load from a range [b, e(
+  // Load impl
   //================================================================================================
-  template<std::input_iterator Iterator>
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), Iterator b, Iterator e) noexcept
-  {
-    using base = typename std::iterator_traits<Iterator>::value_type;
-    return load(ignore_none, safe, as<as_wide_t<base>>{}, b, e);
-  }
 
-  template<std::input_iterator Iterator, typename Wide>
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), as<Wide> const&, Iterator b, Iterator e) noexcept
+  template<callable_options O, std::input_iterator It, typename Wide>
+  EVE_FORCEINLINE Wide load_(EVE_REQUIRES(cpu_), O const&, It src, [[maybe_unused]] It end, as<Wide> tgt) noexcept
   {
-    return load(ignore_none, safe, as<Wide>{}, b, e);
+    if constexpr (logical_simd_value<Wide> && !Wide::abi_type::is_wide_logical)
```

Comment:
There is some confusion here. 

This code is duplicated between this overload and the next overload.
Should be unified for sure.

Reply:
Fixed